lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php

```diff
@@ -2177,12 +2177,19 @@ public function testLength() {
 
 		$result = $this->db->length('decimal(20,3)');
 		$this->assertEquals('20,3', $result);
+	}
 
+/**
+ * Test length parsing of enum column.
+ *
+ * @return void
+ */
+	public function testLengthEnum() {
 		$result = $this->db->length('enum("one", "longer")');
-		$this->assertEquals(6, $result);
+		$this->assertNull($result);
 
 		$result = $this->db->length("enum('One Value','ANOTHER ... VALUE ...')");
-		$this->assertEquals(21, $result);
+		$this->assertNull($result);
```

Comment:
While I understand that the old behavior may not have been entirely correct. That behavior has been there for quite some time and removing it now is pretty risky.

Reply:
Fixing bugs always have some degree of risk. Enum column had to return `null` length in the first place like datetime and text columns do.

Reply:
I agree that fixing bugs could break untested behavior. This seems more deliberate though. Previously the length was the length of the longest accepted element, now it is `null`.

Reply:
And that was a bug. The length is used in defining the column type. So if the length is 15 then the column becomes `'type' => 'enum(15)',` instead of `'type' => 'enum(\'active\',\'deleted\')',`.  Should I add more tests of methods that use `Mysql::length()` method?